### ISSUE TITLE
(D3D)Crash on startup

### ISSUE THREAD

**Issue by kstuedem** [libretro/RetroArch#1646]
37c5c0d98ac4af62acf694a464c2127fecaacc52 replaced new and delete with calloc and free in d3d.cpp, but d3d_video_t contains STL container that aren't properly initialized by calloc in mingw builds.

**Comment by twinaphex** [comment]
Instead of going bak to new/delete (which I really don't want to do, I want to start making it less C++based instead of more), perhaps just initialize std::vector variables specifically?

**Comment by Alcaro** [comment]
There is such a thing as placement new.
```
struct x {
vector<int> member;
};

x* ptr = malloc(sizeof(x));
new(ptr) x();

ptr->~x();
free(ptr);
```
but that's about two thousand times uglier than new. If you're going to de-C++ that file, start with getting rid of the vectors. Until then, new is the best solution.

**Comment by twinaphex** [comment]
OK, change it back to new/delete for new. When I have some more time I'll start on getting rid of the vectors.

**Comment by Alcaro** [comment]
Okay, should be fixed by 79280258c7c8.
